From 228a4f486bd878c032d0956941c4a08472ae47fe Mon Sep 17 00:00:00 2001
From: Artem Senichev <a.senichev@yadro.com>
Date: Sun, 12 Jul 2020 10:27:22 +0300
Subject: [PATCH] Handle big network latency during DHCP requests

By default, busybox DHCP client restarts an IP leasing procedure if a
server does not respond after the third Request message.
This patch adds workaround for such cases by increasing the number of
attempts.

Signed-off-by: Artem Senichev <a.senichev@yadro.com>
---
 networking/udhcp/dhcpc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/networking/udhcp/dhcpc.c b/networking/udhcp/dhcpc.c
index 5b3fd531c..2bc2120b2 100644
--- a/networking/udhcp/dhcpc.c
+++ b/networking/udhcp/dhcpc.c
@@ -1497,7 +1497,7 @@ int udhcpc_main(int argc UNUSED_PARAM, char **argv)
 				packet_num = 0;
 				continue;
 			case REQUESTING:
-				if (packet_num < 3) {
+				if (packet_num < 100) {
 					/* send broadcast select packet */
 					send_select(xid, server_addr, requested_ip);
 					timeout = discover_timeout;
-- 
2.27.0

